ngraph/core/src/pass/make_stateful.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/pass/make_stateful.hpp"
+
+#include <memory>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+#include <openvino/op/util/variable.hpp>
+#include <openvino/opsets/opset8.hpp>
+
+using namespace std;
+using namespace ngraph;
+using namespace opset8;
+using namespace op::util;
+
+namespace {
+string generate_variable_name(const shared_ptr<Parameter>& param, const shared_ptr<Result>& res, uint64_t idx) {
+    return param->get_friendly_name() + res->get_friendly_name() + std::to_string(idx);
+}
+}  // namespace
+
+bool ov::pass::MakeStateful::run_on_function(std::shared_ptr<ngraph::Function> f) {
+    VariableVector variables;
+    SinkVector sinks;
+    uint64_t idx = 0;
+    for (const auto& pair : m_pairs_to_replace) {
+        const auto& param = pair.first;
+        const auto& res = pair.second;
+
+        NGRAPH_CHECK(param->get_partial_shape().is_static(),
+                     "Shape of Parameter ",
+                     param->get_friendly_name(),
+                     " must be static. MakeStateful transformation doesn't support dynamic shapes.");
+
+        const auto& target_inputs = param->get_output_target_inputs(0);
+
+        // Create Variable
+        std::string var_name = generate_variable_name(param, res, idx++);
+        auto variable = std::make_shared<Variable>(VariableInfo{PartialShape::dynamic(), element::dynamic, var_name});
```

Comment:
can we initialize this Variable with actual shape and type?

Reply:
done